lib/pal/PAL.cpp

```diff
@@ -343,19 +343,24 @@ namespace PAL_NS_BEGIN {
 	std::transform(uuidStr.begin(), uuidStr.end(), uuidStr.begin(), ::tolower);
         return uuidStr;
 #else
+        static std::mt19937 random_generator;
```

Comment:
Can you check with Visual Studio debug tracepoint on Windows how often this function gets called on a simple test?

Is it significantly heavier than the old one?

Reply:
std::mt19937 can be heavy. In OpenTelemetry, we use [XorShift](https://en.wikipedia.org/wiki/Xorshift) shift register generator - https://github.com/open-telemetry/opentelemetry-cpp/blob/v1.8.2/sdk/src/common/fast_random_number_generator.h. But better to benchmark.

Reply:
My thoughts are that this spot is NOT used for any encryption-related activities. It would've been risky if we could predict the GUIDs randomly generated, but at massive scale - we haven't yet seen a duplicate GUID, even if the function itself is not "random enough". There are several suggestions on stackoverflow how this could be improved further.

I am not against the safer/nicer/cleaner solution, but the static code analysis warning in this spot is "false positive".

Reply:
Just ran a simple test program to compare the performance of both of the PRNGs (rand() and mt19337) and it seems there is perf impact in latter's case. It seems mt19937 is 4 times slower than rand().
Agree with your comment about rand() fulfilling the requirement here. I think supressing the warning generated here would be the better approach.

Reply:
Also, @maxgolov can you please share the suggestion link you mentioned on the stackoverflow. I can try looking at those as well to see if they fit here.